actions/global_actions.jsx

```diff
@@ -492,21 +501,23 @@ export async function redirectUserToDefaultTeam() {
         }
     }
 
-    if (teams[teamId]) {
+    const team = teams[teamId];
+    if (team) {
         const channelId = BrowserStore.getGlobalItem(teamId);
         const channel = ChannelStore.getChannelById(channelId);
-        if (channel) {
-            redirect(teams[teamId].name, channel.name);
+        let channelName = Constants.DEFAULT_CHANNEL;
+        if (channel && channel.team_id === team.id) {
+            selectChannel(channel.id)(dispatch, getState);
+            channelName = channel.name;
         } else if (channelId) {
             const {data} = await getChannelAndMyMember(channelId)(dispatch, getState);
             if (data) {
-                redirect(teams[teamId].name, data.channel.name);
-            } else {
-                redirect(teams[teamId].name, Constants.DEFAULT_CHANNEL);
+                selectChannel(channelId)(dispatch, getState);
```

Comment:
Setting currentChannelId fix the problem. Others are just code refactor.

Reply:
Same here